api/src/main/java/ai/djl/modality/cv/transform/ToHalf.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.modality.cv.transform;
+
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.translate.Transform;
+
+/** A {@link Transform} that converts the input to float16 if needed. */
+public class ToHalf implements Transform {
```

Comment:
I don't think this is necessary, user can use Lambda function as Transform easily:
```
addTransform(a -> a.toType(DataType.FALOAT16, false);
```

Reply:
I think for simplicity it might be good to have this since the same thing could be said about ToTensor. I can remove it if everyone agrees that it is not needed though.

Reply:
1. We added ToTensor to solve discoverability issue, toType() seems obvious to me
2. It's arguable if we add ToHalf, why we don't add ToUint8, ToFloat64 etc.

Reply:
Both points make sense, and I definitely agree with #2. The main reason I didn't prefer the lambda is due to the check if it should actually convert the type as well as whether it should be a copy or not. I will go ahead and remove it and users can implement their own via lambda or class.